halyard-config/src/main/java/com/netflix/spinnaker/halyard/model/v1/Reference.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.model.v1;
+
+import lombok.Data;
+
+/**
+ * A reference to a class in halconfig.
+ *
+ * @param <T> Type of the reference
+ */
+@Data
+public class Reference<T> {
+  /**
+   * The optional value of the reference. May be null.
+   */
+  private T value;
```

Comment:
It's a good point, but this value just gets passed transparently to one of halconfig fields. Right now I never have to check if it's null, so wrapping it in `Optional` means extra lines of code.